airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py

```diff
@@ -664,6 +666,45 @@ def _flat_associations(self, records: Iterable[MutableMapping]) -> Iterable[Muta
             yield record
 
 
+class SemiIncrementalStream(Stream, IncrementalMixin):
+    _cursor_value = 1
+
+    @property
+    def cursor_field(self) -> Union[str, List[str]]:
+        return self.updated_at_field
+
+    @property
+    def state(self) -> Mapping[str, Any]:
+        return {self.cursor_field: str(self._cursor_value)}
+
+    @state.setter
+    def state(self, value: Mapping[str, Any]):
+        self._cursor_value = value[self.cursor_field]
+
+    def filter_by_state(self, stream_state: Mapping[str, Any] = None, record: Mapping[str, Any] = None) -> bool:
+        record_value = (
+            pendulum.parse(record.get(self.cursor_field)).int_timestamp
+            if isinstance(record.get(self.cursor_field), str)
+            else record.get(self.cursor_field) // 1000
```

Comment:
why 1000?

Reply:
refactored. date-fields  is stored in milliseconds from now.